apps/voting/contracts/Voting.sol

```diff
@@ -361,47 +513,52 @@ contract Voting is IForwarder, AragonApp {
         vote_.executionScript = _executionScript;
 
         emit StartVote(voteId, msg.sender, _metadata);
-
-        if (_castVote && _canVote(voteId, msg.sender)) {
-            _vote(voteId, true, msg.sender);
-        }
     }
 
     /**
-    * @dev Internal function to cast a vote or object to.
-      @dev It assumes that voter can support or object to the vote
-    */
-    function _vote(uint256 _voteId, bool _supports, address _voter) internal {
+     * @dev Internal function to cast a vote or object to.
+     * @dev It assumes that voter can support or object to the vote
+     * @param _voteId The identifier of the vote
+     * @param _supports Whether the voter supports the vote or not
+     * @param _voter The address of the voter
+     * @param _isDelegate Whether the voter is a delegate or not
+     */
+    function _vote(uint256 _voteId, bool _supports, address _voter, bool _isDelegate) internal {
         Vote storage vote_ = votes[_voteId];
 
         // This could re-enter, though we can assume the governance token is not malicious
         uint256 voterStake = token.balanceOfAt(_voter, vote_.snapshotBlock);
         VoterState state = vote_.voters[_voter];
 
         // If voter had previously voted, decrease count
-        if (state == VoterState.Yea) {
+        if (state == VoterState.Yea || state == VoterState.DelegateYea) {
             vote_.yea = vote_.yea.sub(voterStake);
-        } else if (state == VoterState.Nay) {
+        } else if (state == VoterState.Nay || state == VoterState.DelegateNay) {
             vote_.nay = vote_.nay.sub(voterStake);
         }
 
         if (_supports) {
             vote_.yea = vote_.yea.add(voterStake);
-            vote_.voters[_voter] = VoterState.Yea;
+            vote_.voters[_voter] = _isDelegate ? VoterState.DelegateYea : VoterState.Yea;
         } else {
             vote_.nay = vote_.nay.add(voterStake);
-            vote_.voters[_voter] = VoterState.Nay;
+            vote_.voters[_voter] = _isDelegate ? VoterState.DelegateNay : VoterState.Nay;
         }
 
         emit CastVote(_voteId, _voter, _supports, voterStake);
 
         if (_getVotePhase(vote_) == VotePhase.Objection) {
             emit CastObjection(_voteId, _voter, voterStake);
```

Comment:
Do we still need a separate event for it? Maybe let's consider to add `phase` parameter to CastVote event? It depends on frontend / subgraph / tooling tho

Reply:
We've decided to keep these two events as they were since changing them may result in bad backward compatibility.